actions/bump-version/action.yaml

```diff
@@ -33,12 +33,18 @@ runs:
     - name: Bump version
       id: bump-version
       run: |
-        parameters=(--no-commit --no-tag ${{ inputs.release-type }})
+        parameters=(--no-commit --allow-dirty --no-tag ${{ inputs.release-type }})
         echo "old-version=$(python -c "from configparser import ConfigParser; cfg = ConfigParser(); cfg.read('.bumpversion.cfg'); print(cfg['bumpversion']['current_version'])")" >> "$GITHUB_OUTPUT"
-        if [ -n "${{ inputs.new-version }}" ]; then
+        if [[ ${{ inputs.release-type }} == patch ]]; then
           parameters+=(--new-version ${{ inputs.new-version }})
+          bump2version --allow-dirty release
+        else
+          bump2version --allow-dirty "${parameters[@]}"
+          bump2version --allow-dirty release
         fi
-        bump2version "${parameters[@]}"
         echo "new-version=$(python -c "from configparser import ConfigParser; cfg = ConfigParser(); cfg.read('.bumpversion.cfg'); print(cfg['bumpversion']['current_version'])")" >> "$GITHUB_OUTPUT"
+      # We use the case for release-type "patch", because if we do not we get inproper results. For example : 0.0.1-snapshot -> 0.0.3
+      # If release-type == patch , we use "bump2version release". This command basically removes the suffix. For example :  0.0.1-snapshot -> 0.0.1
+      # In the other case we bump the version (major or minor), then we remove the snapshot suffix
```

Comment:
These comments are nice, but they are not quite in the correct place. Could you at them to the fitting if cases that they explain?